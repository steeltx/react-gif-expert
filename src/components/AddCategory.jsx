import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory } ) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        // evitar el refresh por default que se realiza con el form
        event.preventDefault();
        // si no hay caracteres, no hacer nada
        if (inputValue.trim().length <= 1) return;
        // agregar las nuevas categorias a las que ya existen
        //setCategories( categories => [ inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        // una vez ingresado el valor, limpiar el input
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}