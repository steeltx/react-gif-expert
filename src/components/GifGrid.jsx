import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( {category} ) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3> { category } </h3>
            {
                isLoading && (<h2>Cargando ...</h2>)
            }
            
            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        <GifItem 
                            key={image.id}
                            { ...image } // enviar todas las propiedades de image al componente
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}