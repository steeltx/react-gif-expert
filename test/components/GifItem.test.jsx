import { render, screen } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {

    const title = 'Saitama';
    const url = 'https://url-ejemplo';

    test('debe de hacer match con el snapshot', () => {
        // renderizar el componente y obtener el container para evaluar
        const  { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con la URL y el ALT indicado', () => {
        render( <GifItem title={title} url={url} /> );
        // mostrar lo que se esta renderizando
        // screen.debug();

        //expect(screen.getByRole('img').src).toBe(url+'/');
        //expect(screen.getByRole('img').alt).toBe(title);

        // obtener los valores a evaluar en una sola linea con desestructuracion
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url+'/');
        expect(alt).toBe(alt);

    });

    test('debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={title} url={url} /> );
        expect(screen.getByText(title)).toBeTruthy();
    });

});