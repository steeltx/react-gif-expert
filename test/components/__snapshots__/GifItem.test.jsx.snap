// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://url-ejemplo"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
